test: allow networking in testlib's global machine

larskarlitski opened the pull request:
The global machine is created when running non-destructive tests without
passing `--machine`. It is created without networking and there is no
way for a test to override this default.

The machines we run non-destructive tests against usually have
networking enabled. Thus, enable networking for the global machine as
well.

martinpitt commented:
This would (potentially) break Cockpit's tests, they really should run offline. How about adding a boolean argument to `get_global_machine()`, and exposing it in `run-tests` as a command line option?

larskarlitski commented:
> This would (potentially) break Cockpit's tests, they really should run offline. How about adding a boolean argument to `get_global_machine()`, and exposing it in `run-tests` as a command line option?

Ah, I see. Is networking disabled when running in gating as well then?

Passing to `run-tests` isn't enough, because these we need network when running `test/verify/check-*` directly. How about setting a global variable on `MachineCase` that can be overriden from a test?

larskarlitski commented:
Thanks @martinpitt! I've changed it according to your suggestion and added another commit to add a `--enable-network` command line argument. This is because `test/common/run-tests` creates its own virtual machines which are unaffected by the `restrict` option.

Maybe always explicitly enabling networking is better anyway?

martinpitt commented on test/common/testlib.py:
```diff
@@ -1528,6 +1528,8 @@ def arg_parser(enable_sit=True):
                             help="Sit and wait after test failure")
     parser.add_argument('--nonet', dest="fetch", action="store_false",
                         help="Don't go online to download images or data")
+    parser.add_argument('--enable-network', dest='enable_network', action='store_true',
+                        help="Enable network access for tests")
```
I'm not sure how useful that's actually going to be -- tests should be able to declare themselves that they need network access, i. e. call `MachineCase.setUp()` accordingly. But it's a quick fix for composer's run-tests invocation indeed, and it doesn't hurt for now.

larskarlitski replied:
hm no, this is actually only for cockpit's run-tests, which creates the global machine itself.

larskarlitski commented:
> Thanks! Can you please add a `Closes #` to the second commit?

Added. Thanks!

larskarlitski commented:
Ignoring `fedora-31/firefox` as per advice from irc. It is currently broken and not relevant to the changes in this PR.
